packages/wallet/ui/src/components/ConnectionSettingsDialog.jsx

```diff
@@ -365,7 +377,8 @@ const ConnectionSettingsDialog = ({
           onClick={saveAndClose}
           disabled={
             errors.size > 0 ||
-            JSON.stringify(config) === JSON.stringify(connectionConfig)
+            (JSON.stringify(config) === JSON.stringify(connectionConfig) &&
+              config.smartConnectionMethod !== SmartConnectionMethod.KEPLR)
```

Comment:
this checks looks coded about three times. maybe worth a small helper, if only for readability

Reply:
Done